src/main/java/io/shiftleft/fuzzyc2cpg/parser/shared/builders/TemplateAstBuilder.java

```diff
@@ -0,0 +1,20 @@
+package io.shiftleft.fuzzyc2cpg.parser.shared.builders;
```

Comment:
This superclass was introduced as both functions _and_ class definitions can have a template definition. This simply abstracts over that so we can generically add template parameters in the `CModuleParserTreeListener`.